### PR TITLE
feat. build sealos cloud cluster image

### PR DESCRIPTION
**Pull request by lingdie**
### Summary
📱🛠️🌐

Moved all files under `deploy/cloud` to `deploy/cloud-deprecated` to mark them as obsolete and prepare for removal. These files were related to deploying sealos on cloud platforms using ingress-nginx and apisix.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Rename the entire `deploy/cloud` directory to `deploy/cloud-deprecated` to indicate that it is no longer maintained or supported ([link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-d2a1bda0bc3bd485e273a86aca0664a5381594de65f234d3485c38aa0edfb74c)-[link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-359cb62d6e37914f563b05427b07394792305c27d53f2e1fc68fde14c34919ff))
*  Update the README files in the renamed directory to reflect the deprecation status and provide alternative instructions for deploying the cloud components ([link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-d2a1bda0bc3bd485e273a86aca0664a5381594de65f234d3485c38aa0edfb74c), [link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-3795b28dabebff22a5bbacf66409d5a8696511405183064950d8b3b7a2435e69), [link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-797399c735dab58452ed7ee169e9eaf4190a63ad7901fe673a50bcb833bfc36f), [link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-89283ae6f46c25c5519b31a0fcd7060c3e41f73691e708e36e0761e7c75a8528), [link](https://github.com/labring/sealos/pull/3057/files?diff=unified&w=0#diff-359cb62d6e37914f563b05427b07394792305c27d53f2e1fc68fde14c34919ff))

### CODE REVIEW

File: deploy/cloud/Readme.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Sealos cloud cluster image
+
+## usage
+
+### prepare
+
+1. A cloud domain and dns to your k8s cluster. Suppose your domain name is `cloud.example.io`.
+2. A TLS cert that resolves `cloud.example.io` and `*.cloud.example.io`
```
Comment (fanux) [inline deploy/cloud/Readme.md:8]: How to get or generate it?

Comment (fanux) [reply deploy/cloud/Readme.md:8]: Provide examples using public DNS and private DNS.

File: deploy/cloud/Readme.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  data: |
+    data:
+      tls.crt: <your-tls.crt-base64>
+      tls.key: <your-tls.key-base64>
+```
+
+### run sealos cloud cluster image
+```shell
+sealos run ghcr.io/labring/sealos-cloud:dev --config-file tls-secret.yaml --env cloudDomain="cloud.example.com"
+```
```
Comment (fanux) [inline deploy/cloud/Readme.md:31]: How to cname or use `/etc/hosts` in some POC demo case?

Comment (lingdie) [reply deploy/cloud/Readme.md:31]: Due to the current architecture design and code reasons, deployment without a valid certificate is not supported.

Comment (zzjin) [reply deploy/cloud/Readme.md:31]: `cname` and/or `etc/hosts` is controlled by outside handlers such as `godaddy`/`alidns` etc, AIO script can do nothing but a deployment pre-requirement doc, with examples how to cname domain and how to obtain free certs.

File: deploy/cloud/Readme.md (hunk 1 above)
Comment (zzjin) [reply deploy/cloud/Readme.md:8]: May be add docs to links how to issue wildcard cert: https://zerossl.com/features/certificates/
But as install scripts, we cannot help users to auto issue dns-scope cert.

**Comment by cuisongliu** [comment]
!! all in one image replace using `sealos run xxx.tar` 

user not care registry addr and  registry user and registry password

**Comment by lingdie** [comment]
> !! all in one image replace using `sealos run xxx.tar`
> 
> user not care registry addr and registry user and registry password

I will set the registry addr/u/p as env soonsoon latter latter.

**Comment by cuisongliu** [comment]
```
ROM alpine:latest as builder
COPY cri cri
RUN ls -l

FROM alpine:latest
WORKDIR /
COPY --from=builder cri/image-cri-shim .
```

 sealos build -t aaaa --isolation=chroot  .

**Comment by lingdie** [comment]
redirect to pr https://github.com/labring/sealos/pull/3082

**Pull request closed without merging**